src/routers/handlers/personal-code/personalCodeHandler.ts

```diff
@@ -95,4 +100,15 @@ export class PersonalCodeHandler extends GenericHandler<PersonalCodeViewData> {
         };
     }
 
+    private resolveNextPageUrl (validationStatusResponse: Resource<ValidationStatusResponse>): string {
+        let url = PrefixedUrls.INDIVIDUAL_STATEMENT;
+
+        if (validationStatusResponse.resource && validationStatusResponse.resource.isValid === false) {
+            const hasNameMismatchError = validationStatusResponse.resource.errors?.some((validationError: { error: string | string[]; }) => validationError.error.includes("name mismatch"));
+            url = hasNameMismatchError ? PrefixedUrls.NAME_MISMATCH : PrefixedUrls.INDIVIDUAL_STATEMENT;
```

Comment:
I wonder about the reliability of checking for `"name mismatch"`?  I realise this is what our API returns, but it feels brittle if we were to amend the error message for customers and this no longer matched?  Perhaps we can all discuss if there is any option, but maybe the API ought to prefix an error with some general value??

Reply:
Yes, I agree, but how do we resolve? Happy to discuss.

Reply:
Given this may require changes in API and in Web, I would think it should be a separate story.  This PR works and is based on what is provided at the moment.
The story needn't prescribe a specific solution straight away.